pool/connection_pool.go

```diff
@@ -334,50 +333,34 @@ func (p *ConnectionPool) Close() []error {
 // for all requests to complete.
 func (p *ConnectionPool) CloseGraceful() []error {
 	if p.state.cas(connectedState, shutdownState) {
-		p.addrsMutex.RLock()
-		for _, s := range p.addrs {
+		p.endsMutex.RLock()
+		for _, s := range p.ends {
 			s.cancel()
 			close(s.shutdown)
 		}
-		p.addrsMutex.RUnlock()
+		p.endsMutex.RUnlock()
 	}
 
 	return p.waitClose()
 }
 
-// GetAddrs gets addresses of connections in pool.
-func (p *ConnectionPool) GetAddrs() []string {
-	p.addrsMutex.RLock()
-	defer p.addrsMutex.RUnlock()
-
-	cpy := make([]string, len(p.addrs))
-
-	i := 0
-	for addr := range p.addrs {
-		cpy[i] = addr
-		i++
-	}
+// GetInfo gets information of connections (connected status, ro/rw role).
+func (p *ConnectionPool) GetInfo() map[string]ConnectionInfo {
+	info := make(map[string]ConnectionInfo)
 
-	return cpy
-}
-
-// GetPoolInfo gets information of connections (connected status, ro/rw role).
-func (p *ConnectionPool) GetPoolInfo() map[string]*ConnectionInfo {
-	info := make(map[string]*ConnectionInfo)
-
-	p.addrsMutex.RLock()
-	defer p.addrsMutex.RUnlock()
+	p.endsMutex.RLock()
+	defer p.endsMutex.RUnlock()
 	p.poolsMutex.RLock()
 	defer p.poolsMutex.RUnlock()
 
 	if p.state.get() != connectedState {
 		return info
 	}
 
-	for addr := range p.addrs {
-		conn, role := p.getConnectionFromPool(addr)
+	for id := range p.ends {
+		conn, role := p.getConnectionFromPool(id)
 		if conn != nil {
-			info[addr] = &ConnectionInfo{ConnectedNow: conn.ConnectedNow(), ConnRole: role}
+			info[id] = ConnectionInfo{ConnectedNow: conn.ConnectedNow(), ConnRole: role}
 		}
```

Comment:
```suggestion
		} else {
		     info[id] = ConnectionInfo{ConnectedNow: false, ConnRole: RoleUnknown}
		}
```

Reply:
I return pointer return type back, so we don't need `else`.

Reply:
Return `type`.